state/src/impls/top_level.rs

```diff
@@ -149,6 +150,72 @@ impl TopStateView for TopLevelState {
     }
 }
 
+impl DbContext for TopLevelState {
+    fn get(&self, storage_id: StorageId, key: &DbCxtKey) -> Option<DbCxtValue> {
+        let error_handler = |e| {
```

Comment:
The similar definitions of `error_handler` are repeated. I think it can be written with a macro.
```rust
macro_rules! panic_at {
    ($method: expr, $e: expr) => {
        panic!("DbContext {} method failed with {}", $method, $e);
    };
}
```

Reply:
I applied the macro